Bottom: b451f32fb1fdc37a514289f58a1e8aa08c444bfe
Top:    e5c30dde4c292d3cfa5d71faca70f8a4b7c6a657
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 17:14:44 +0000

Refactor polygon copy operations


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9282eda..19ec771 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2991,13 +2991,13 @@ BOOLp
 poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
 {
   *dst = NULL;
-  if (src != NULL)
-    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if (src == NULL)
+    return TRUE;
+
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
